### PR TITLE
Initial testing for the mariadb container

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Review by mloviska — APPROVED** [review]

**Review by dcermak — CHANGES_REQUESTED** [review]
`test_build` and `test_all` are broken

File: bci_tester/data.py [hunk 1]
```diff
@@ -542,6 +542,25 @@ def create_BCI(
...
+MARIADB_ROOT_PASSWORD = "n0ts3cr3t"
+
+MARIADB_CONTAINERS = [
+    create_BCI(
+        build_tag=f"{APP_CONTAINER_PREFIX}/mariadb:{mariadb_ver}",
+        bci_type=ImageType.APPLICATION,
+        available_versions=maria_versions,
+        forwarded_ports=[PortForwarding(container_port=3306)],
+        extra_environment_variables={
+            "MARIADB_PASSWORD": MARIADB_ROOT_PASSWORD
```
Comment (dcermak) [inline bci_tester/data.py:554]:
```suggestion
            "MARIADB_ROOT_PASSWORD": MARIADB_ROOT_PASSWORD
```

**Review by mgrossu — APPROVED** [review]
LGTM

File: tests/test_mariadb.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    cur = None
+
+    dbdir = "/var/lib/mysql"
+
+    dbdir_f = container_per_test.connection.file(dbdir)
+    assert dbdir_f.exists
+    assert dbdir_f.user == "mysql"
+    assert dbdir_f.mode == 0o755
+
+    assert container_per_test.connection.check_output("id -un") == ("root")
```
Comment (ricardobranco777) [inline tests/test_mariadb.py:101]:
```suggestion
    assert container_per_test.connection.check_output("id -un") == "root"
```

**Review by ricardobranco777 — APPROVED** [review]
Small suggestion. LGTM

**Comment by dirkmueller** [comment]
(need to work on docker failures)

File: tests/test_mariadb.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+from bci_tester.data import MARIADB_CONTAINERS
+from bci_tester.data import MARIADB_ROOT_PASSWORD
+
+
+CONTAINER_IMAGES = MARIADB_CONTAINERS
+
+
+def test_entry_point(auto_container: ContainerData) -> None:
+    # do really nothing here, just check that the container launched
+    assert auto_container.connection.run_expect([0], "ps")
```
Comment (dcermak) [inline tests/test_mariadb.py:25]: You don't really need this, if the container does not launch, then the test setup will fail.

**Pull request merged**